pkg/ddl/notifier/store.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notifier
+
+import (
+	"context"
+	"fmt"
+
+	sess "github.com/pingcap/tidb/pkg/ddl/session"
+)
+
+// Store is the (de)serialization and persistent layer.
+type Store interface {
+	Insert(context.Context, *sess.Session, *schemaChange) error
+	UpdateProcessed(
+		ctx context.Context,
+		se *sess.Session,
+		ddlJobID int64,
+		multiSchemaChangeID int,
+		processedBy uint64,
+	) error
+	Delete(ctx context.Context, se *sess.Session, ddlJobID int64, multiSchemaChangeID int) error
+	List(ctx context.Context, se *sess.Session, limit int) ([]*schemaChange, error)
+}
+
+// DefaultStore is the system table store. Still WIP now.
+var DefaultStore Store
+
+type tableStore struct {
+	db    string
+	table string
+}
+
+func (t *tableStore) Insert(ctx context.Context, s *sess.Session, change *schemaChange) error {
+	// TODO: fill schema_change after we implement JSON serialization.
+	sql := fmt.Sprintf(`
```

Comment:
Do we need to use the parameterized queries? Usually, it would be:
```go
func (t *tableStore) Insert(ctx context.Context, s *sess.Session, change *schemaChange) error {
    sql := `
        INSERT INTO ?.? (
            ddl_job_id,
            multi_schema_change_seq,
            schema_change,
            processed_by_flag
        ) VALUES (?, ?, ?, 0)`
    
    _, err := s.Execute(ctx, sql, "ddl_notifier", t.db, t.table, change.ddlJobID, change.multiSchemaChangeSeq, "{}")
    return err
}
```

Reply:
The SQL string interpolation is also done in our codebase in `EscapeSQL`, I think we use this specialized printf is equivalent. And `EscapeSQL` can not avoid SQL injection as its comment said, generally it will not improve security.

Reply:
Got it.